CONTRIBUTING.md

```diff
@@ -139,6 +139,47 @@ If you haven't sent in pull requests before, [here is GitHub's documentation](ht
 
 For PRs to be accepted, they go through a review process. Generally there is a feedback cycle where someone reviews and requests some changes. All PRs need to pass our tests. It is also good to send in new code with test coverage.
 
+## Merging Pull Requests
+
+Pull Requests should be merged with either option `Create a merge commit` or
+`Squash and merge`, but __not__ with `Rebase and merge`.
+
+### Create a merge commit
+
+This option keeps the pull request's full history, with all the authors information
+and all individual commits. This is the best default choice, and the recommended
+choice if the PR is composed of more than 1 commit and they're all well split up
+in logical chunks.
+
+When creating a merge commit, it's important that you change the merge commit's
+title. It should read something like:
+
+> <Short description of what changes in this PR, can often be the PR title> (Merge PR #XXX)
+
+Note the words `Merge PR` before the pull request number.
+
+You can change the title if necessary, and you can also add more information to
+the commit message, and possibly add `Fixes #XXX` if necessary.
+
+### Squash and merge
+
+This option will put all commits in the pull request into one single commit,
+and commit this single commit only to the main branch. We'll loose all author
+information but one. This is recommended if the PR is composed of only one
+commit, or several commits that aren't independent (which can be the case when
+the contributor is new to the git system).
+
+When creating a squashed commit, github will automatically use the pull request
+title as the commit title, but please edit it where needed. It should look like
+this:
+
+> <Short description of what changes in this PR, can often be the PR title> (PR #XXX)
+
+Note the word `PR` before the pull request's number.
+
+Again you can edit the commit log if necessary, and possibly add `Fixes #XXX` if
+necessary.
```

Comment:
I feel like we can remove one `if necessary`. How about:
```suggestion
Again, you can edit the commit log, and possibly add `Fixes #XXX` if necessary.
```

Reply:
Ah yeah there's clearly one too many, good catch! But I'll remove the second one, it sounds better to me.

Reply:
I also rewrote a bit the similar sentence in the previous paragraph, there was the same problem.